### PR TITLE
test:fix plugin tests

### PR DESCRIPTION
**Pull request by chloeyin**
1. Rewrite plugin test

Signed-off-by: zaihaoyin <zaihaoyin@microsoft.com>

### CODE REVIEW

File: signature/plugin_test.go [hunk 1]
```diff
@@ -2,425 +2,836 @@ package signature
...
+	validMetaDataWithSigningCapabilityFunc = func(ctx context.Context, req plugin.Request) (interface{}, error) {
+		return &validMetadata, nil
+	}
+	validMetaDataWithEnvelopeGeneratorCapabilityFunc = func(ctx context.Context, req plugin.Request) (interface{}, error) {
+		metaData := validMetadata
+		metaData.Capabilities = []plugin.Capability{plugin.CapabilityEnvelopeGenerator}
+		return &metaData, nil
+	}
+	// TODO: plugin spec may change
+	// validDescribeKeyWFunc = func(ctx context.Context, req plugin.Request) (interface{}, error) {
```
Comment (binbin-li) [inline signature/plugin_test.go:51]: do we need to remove it?

File: signature/plugin_test.go [hunk 2]
```diff
@@ -2,425 +2,836 @@ package signature
...
+	signFunc
+	certChainFunc
+	keySpecFunc
+	runner *mockRunner
 }
 
-func (r *mockRunner) Run(_ context.Context, _ plugin.Request) (interface{}, error) {
-	defer func() { r.n++ }()
-	return r.resp[r.n], r.err[r.n]
+type option func(*options)
```
Comment (binbin-li) [inline signature/plugin_test.go:63]: nit: I feel `optionFunc` might make more sense

File: signature/plugin_test.go (hunk 1 above)
Comment (chloeyin) [reply signature/plugin_test.go]: > do we need to remove it?

Removed

File: signature/plugin_test.go (hunk 2 above)
Comment (chloeyin) [reply signature/plugin_test.go]: Updated

**Pull request merged**